packages/cli-kit/src/public/node/node-package-manager.ts

```diff
@@ -504,8 +504,9 @@ function argumentsToAddDependenciesWithYarn(dependencies: string[], type: Depend
  * @returns An array with the arguments.
  */
 function argumentsToAddDependenciesWithPNPM(dependencies: string[], type: DependencyType): string[] {
-  let command = ['add']
+  let command = ['add', '-w']
```

Comment:
@matteodepalo  I have tophated this and there is an error when you try to run the  `upgrade` command if the app is not configured to use `workspaces`. I guess we should only add this flag in case the app use that feature.

![image](https://github.com/Shopify/cli/assets/105213827/dc738364-8bf5-4a43-a81a-b3844515d0d5)

Reply:
Good catch I'll add this check.